src/newsletter-snackbar/newsletter-snackbar.js

```diff
@@ -15,6 +14,10 @@ export default function NewsletterSnackbar() {
 
     setOpen(false);
   };
+  useEffect(() => {
+    if (referredFromNewsletterSignup)
+      setItem(items.isSubscribedToNewsletter, true);
+  });
```

Comment:
This effect shouldn't be run after every render. Add an empty array of dependencies as a second argument so it's only run on mount/unmount.